lib/dns_wrapper.rb

```diff
@@ -12,46 +12,20 @@ def dns_client
   
   public
   
-  def lookup_dns(zone_id, domain_name)
-    aws_ip = lookup_dns_a_record(zone_id, domain_name)
-    dns_ip = Resolv.getaddress(domain_name)
-    fail("Discrepancy for #{domain_name}: AWS=#{aws_ip} but DNS=#{dns_ip}") unless dns_ip == aws_ip
-    dns_ip
-  end
-  
-  def update_dns_a_record(zone_id, domain_name, new_ip)
-    update_response = request_update_dns_a_record(zone_id, domain_name, new_ip)
-    
-    1.upto(WAIT_ATTEMPTS) do |try|
-      break if update_insync?(update_response)
-      fail('Giving up') if try >= WAIT_ATTEMPTS
-      LOGGER.info("try #{try}: DNS update not yet propagated to AWS nameservers...")
-      sleep(WAIT_INTERVAL)
-    end
-    
-    1.upto(WAIT_ATTEMPTS) do |try|
-      break if Resolv.getaddress(domain_name) == new_ip
-      fail('Giving up') if try >= WAIT_ATTEMPTS
-      LOGGER.info("try #{try}: DNS update not yet propagated to local nameserver...")
-      sleep(WAIT_INTERVAL)
+  def lookup_cname(zone_id, name)
+    cname_from_dns = Resolv::DNS.new.getresource(name, Resolv::DNS::Resource::IN::CNAME).name.to_s
+    cname_from_aws = lookup_dns_cname_record(zone_id, name)
+    if cname_from_dns != cname_from_aws
+      fail("CNAME from DNS (#{cname_from_dns}) != CNAME from AWS (#{cname_from_aws})")
     end
+    cname_from_aws
   end
```

Comment:
Would we want to block ourselves from making a swap until changes have propagated to whichever DNS server is returning dig requests for us? Seems like a bit of an arbitrary contraint, unless I'm missing something. If AWS response didn't match a dig response, but we go ahead with the CNAME switch anyway, what harm would come of that?